src/Authenticator/ChangePasswordHandler.php

```diff
@@ -0,0 +1,190 @@
+<?php declare(strict_types=1);
+
+namespace SilverStripe\MFA\Authenticator;
+use LogicException;
+use Psr\Log\LoggerInterface;
+use SilverStripe\Control\HTTPRequest;
+use SilverStripe\Control\HTTPResponse;
+use SilverStripe\Core\Injector\Injector;
+use SilverStripe\MFA\RequestHandler\BaseHandlerTrait;
+use SilverStripe\MFA\RequestHandler\LoginHandlerTrait;
+use SilverStripe\MFA\Service\MethodRegistry;
+use SilverStripe\MFA\Service\SchemaGenerator;
+use SilverStripe\MFA\Store\StoreInterface;
+use SilverStripe\Security\Member;
+use SilverStripe\Security\MemberAuthenticator\ChangePasswordHandler as BaseChangePasswordHandler;
+
+/**
+ * Extends the "MemberAuthenticator version of the ChangePasswordHandler in order to allow MFA to be
+ * inserted into the flow when an AutoLoginHash is being used  - that is when the user has clicked a
+ * reset password link in an email after using the "forgot password" functionality.
+ * When an "auto login" is not being used (a user is already logged in), it is existing functionality
+ * to ask a user for their password before allowing a change - so this flow does not require MFA.
+ */
+class ChangePasswordHandler extends BaseChangePasswordHandler
+{
+    use BaseHandlerTrait;
+    use LoginHandlerTrait;
+
+    const MULTIFACTORAUTHENTICATED = 'MultiFactorAuthenticated';
```

Comment:
I have a minor gripe with this in that it doesn't tell you what it's for - it's a session key right? Can we call it `SESSION_KEY` instead or something similar?